pandas/io/pytables.py

```diff
@@ -4022,7 +4022,11 @@ def read(self, where=None, columns=None, **kwargs):
             df = concat(frames, axis=1, verify_integrity=False).consolidate()
 
         # apply the selection filters & axis orderings
-        df = self.process_axes(df, columns=columns)
+        if type(columns) is list:
```

Comment:
Checking for specific types is generally a bad idea -- what if, for example, a user passes in a list subclass, or another sequence type?

A better approach is to cast any provided columns to a list:
```python
if columns is not None:
    columns = list(columns)
```

However, given that the mutation happens in the `process_axes` method, it would be a better idea to put it there.